src/openapi/authProvider.ts

```diff
@@ -92,14 +92,24 @@ type Options = {
     maxRetryCount?: number;
 };
 
+const EVENT_TOKEN_REFRESH = 'tokenRefresh' as const;
+const EVENT_TOKEN_RECEIVED = 'tokenReceived' as const;
+const EVENT_TOKEN_REFRESH_FAILED = 'tokenRefreshFailed' as const;
+
+type EmittedEvents = {
+    [EVENT_TOKEN_REFRESH]: () => void;
+    [EVENT_TOKEN_RECEIVED]: (token?: string, refresh?: number) => void;
+    [EVENT_TOKEN_REFRESH_FAILED]: () => void;
+};

```

Comment:
This is the way we tell TS what events are supported - what arguments we can trigger and what arguments we can receive in on/one/off callbacks. My initial implementation was based on tuples so the structure looked like this:
```
{
    eventName: [arg1: string, arg2: number]
}
```
but I decided to expect a typed callback since it seems more clear.